src/include/ipc/topology.h

```diff
@@ -182,9 +182,16 @@ struct sof_ipc_comp_process {
 	struct sof_ipc_comp_config config;
 	uint32_t size;	/**< size of bespoke data section in bytes */
 	uint32_t type;	/**< sof_ipc_effect_type */
+	uint32_t obs;   /**< output buffer size - free space in output buffer
+			  *  measured in samples required to run module
+			  */
```

Comment:
Objection: The size in samples will depend of the actual sampling rate which could come later as a hw_param. Also some processing may generate stuff that is not based on samples in the first place.

Reply:
not to mention that even if a processing a specific buffer size requirements, e.g. 256 samples for an FFT, others may just work fine with whatever buffer is provided to them if they work at the sample level, so how would this latter case be handled. We'd need a flag that says don't care.

Reply:
@plbossart Regarding parameters coming later as hw_params I have following assumptions:
1. As topology is defined the ibs/obs are set for default size (default audio format). There is an assumption that there is enough memory to support worst case if hw_params changes audio format.
2. In run-time hw_params may update audio processing format triggering component to update it's ibs/obs.
3. Ultimately as driver will have ability to destroy pipelines in run-time typical flow will be via create / set hw_params /run / destroy sequence.
4. Is @bkokoszx comment regarding obs/ibs set to 0 as indication that component can work with variable amount of data acked ?

Reply:
Can the naming from obs and ibs be improved ? e.g obs could be min_sink_frames
This way far more meaning can be derived from reading the usage in the code.

Reply:
@lgirdwood 
I've just changed the naming from ```ibs``` and ```obs``` to ```min_source_samples``` and ```min_sink_samples```, since ```ibs``` and ```obs``` was measured in samples.